python_modules/dagster/dagster_tests/declarative_automation_tests/daemon_tests/test_e2e.py

```diff
@@ -753,3 +753,31 @@ def test_observable_source_asset() -> None:
             _execute_ticks(context, executor)  # pyright: ignore[reportArgumentType]
             runs = _get_runs_for_latest_ticks(context)
             assert len(runs) == 0
+
+
+def test_observable_source_asset_is_not_backfilled() -> None:
+    with (
+        get_grpc_workspace_request_context("hourly_observable_with_partitions") as context,
+        get_threadpool_executor() as executor,
+    ):
+        asset_graph = context.create_request_context().asset_graph
+
+        time = datetime.datetime(2024, 8, 16, 1, 35)
+        with freeze_time(time):
+            _execute_ticks(context, executor)  # pyright: ignore[reportArgumentType]
+            backfills = _get_backfills_for_latest_ticks(context)
+            assert len(backfills) == 0
+
+        time += datetime.timedelta(hours=1)
+        with freeze_time(time):
+            _execute_ticks(context, executor)  # pyright: ignore[reportArgumentType]
+            backfills = _get_backfills_for_latest_ticks(context)
```

Comment:
we should also make sure that a run gets kicked off for the observable source asset here

Reply:
Should it be just a run for the observable source asset? I think it gets kicked off, but a run for the materializable asset doesn't get kicked off, and that was confusing me...

Reply:
yep basically:

- 1 run for the observable source asset
- 1 backfill for the materializable asset

that backfill will end up kicking off runs, but the queries in the utility functions there are scoped to things directly launched by the latest tick, and so would ignore any runs spawned from the backfill

Reply:
> the queries in the utility functions there are scoped to things directly launched by the latest tick, and so would ignore any runs spawned from the backfill

Makes sense, I didn't understand this initially!